### PR TITLE
DOC: Negative binomial distribution clarification

### PR DESCRIPTION
**Pull request by Kornel**
The negative binomial distribution can be parametrized with either
the number of succeses or failures. The tutorial explains what was
chosen in the scipy implementation, but the reference does not.

 In my opinion it's worth to describe it here, to me it is a must have
to be able to look it up quickly in the reference.

### CODE REVIEW

**Comment by ev-br** [comment]
This is certainly a very good idea, thanks for working on this!

The wording however could be improved. First of all, these are not necessarily keywords, positional arguments are also accepted. I'd suggest something along the lines of "Negative binomial distribution describes a sequence of i.i.d. Bernoulli trials, repeated until a predefined, non-random number of failures occurs. `nbinom` is parameterized via the number of successes ``n`` and a probability of a single success ``p``". And it should go into the Notes section.

**Comment by josef-pkt** [comment]
@ev-br  your sentence sounds a bit mixed up
either number of successes or failures is fixed, the other one is the random variable
e.g. if successes is fixed at parameter `n`, then k is the number of failures
AFAIU

(In statistics I'm more used to the interpretation that it is a Poisson distribution with a gamma mixture for poisson/mean parameter.)

**Comment by Kornel** [comment]
@ev-br Ad keywords: The doc for the [normal distribution](http://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html) talks about keywords, by the way, aren't they positional args as well? 

Another thing: scale and location for the normal dist are described not in the notes sections. Which way should it be - if Notes, I can change the doc for the normal distribution as well (to make it consistent).

@josef-pkt Your perspective (Poisson distr with gamma mixture) got me thinking that maybe the best documentation is the bare pmf. That's what I enjoy about maths in general - everyone has their own intuition of what certain mathematical concepts 'mean' but at the end of the day there is just one formula to be shared amongst people :)

What do you think about such wording, now in the notes section we have:

```
The probability mass function for nbinom is:

nbinom.pmf(k) = choose(k+n-1, n-1) * p**n * (1-p)**k

for k >= 0.

nbinom takes n and p as shape parameters.

```

I suggest:


```
Negative binomial distribution describes a sequence of i.i.d. Bernoulli trials, repeated until a predefined, non-random number of failures occurs. 

The probability mass function of the number of failures for nbinom is:

nbinom.pmf(k) = choose(k+n-1, n-1) * p**n * (1-p)**k

for k >= 0.

nbinom takes n and p as shape parameters where n is the number of successes, whereas p is the probability of a single success"

**Comment by ev-br** [comment]
@Kornel ping. It'd be nice to follow up: if you add a commit to your local branch and push, it'd show up here in the PR. Your wording with Josef's correction (number of failures is not fixed) sounds fine.

**Comment by Kornel** [comment]
@ev-br there you go, I've changed the doc and merge master.

File: scipy/stats/_discrete_distns.py [hunk 1]
```diff
@@ -142,13 +142,16 @@ class nbinom_gen(rv_discrete):
 
     Notes
     -----
-    The probability mass function for `nbinom` is::
+    Negative binomial distribution describes a sequence of i.i.d. Bernoulli trials, 
+    repeated until a predefined, non-random number of failures occurs.
```
Comment (ev-br) [inline scipy/stats/_discrete_distns.py]: This parametrization, as far as I can see (11:20pm local time so please double-check) corresponds to the series of iid Bernoulli trials repeated until a predefined, non-random number of *successes*, occurs.

This is parametrization 3 in https://en.wikipedia.org/wiki/Negative_binomial_distribution#Alternative_formulations

File: scipy/stats/_discrete_distns.py [hunk 2]
```diff
@@ -142,13 +142,16 @@ class nbinom_gen(rv_discrete):
...
 
-         nbinom.pmf(k) = choose(k+n-1, n-1) * p**n * (1-p)**k
+    The probability mass function of the number of failures for `nbinom` is::
+
+    nbinom.pmf(k) = choose(k+n-1, n-1) * p**n * (1-p)**k
 
     for ``k >= 0``.
 
-    `nbinom` takes ``n`` and ``p`` as shape parameters.
+    `nbinom` takes ``n`` and ``p`` as shape parameters where n is the number of successes, whereas p is the probability of a single success.
```
Comment (ev-br) [inline scipy/stats/_discrete_distns.py]: Please keep the line lengths <80 chars.

**Comment by ev-br** [comment]
@Kornel thanks! Some more work still, I've left a couple of comments. Once you push, it'd be great if you rebase instead of merging master and force-push.

**Comment by Kornel** [comment]
@ev-br Done :)

File: scipy/stats/_discrete_distns.py (hunk 1 above)
Comment (Kornel) [inline scipy/stats/_discrete_distns.py]: I did double check and I think you're right :)

**Comment by ev-br** [comment]
Thanks @Kornel ans congrats with what I believe is your first scipy commit. Keep them coming!

**Comment by Kornel** [comment]
Thanks :) it is, I'll do my best.

**Pull request merged**